packages/features/ee/payments/api/webhook.ts

```diff
@@ -174,10 +259,56 @@ async function handlePaymentSuccess(event: Stripe.Event) {
   });
 }
 
+const handleSetupSuccess = async (event: Stripe.Event) => {
+  const setupIntent = event.data.object as Stripe.SetupIntent;
+  const payment = await prisma.payment.findFirst({
+    where: {
+      externalId: setupIntent.id,
+    },
+  });
+
+  if (!payment?.data || !payment?.id) throw new HttpCode({ statusCode: 204, message: "Payment not found" });
+
+  const { booking, user, evt, eventTypeRaw } = await getBooking(payment.bookingId);
+
+  const bookingData: Prisma.BookingUpdateInput = {
+    paid: true,
+  };
+
+  if (!eventTypeRaw?.requiresConfirmation) {
+    const eventManager = new EventManager(user);
+    const scheduleResult = await eventManager.create(evt);
+    bookingData.references = { create: scheduleResult.referencesToCreate };
+    bookingData.status = BookingStatus.ACCEPTED;
+  }
+
+  await prisma.payment.update({
+    where: {
+      id: payment.id,
+    },
+    data: {
+      data: {
+        ...(payment.data as Prisma.JsonObject),
+        setupIntent: setupIntent as unknown as Prisma.JsonObject,
+      },
+      booking: {
+        update: {
+          ...bookingData,
+        },
+      },
+    },
+  });
+
+  // If the card information was already captured in the same customer. Delete the previous payment method
+
+  if (!eventTypeRaw?.requiresConfirmation) await sendScheduledEmails({ ...evt });
```

Comment:
As part of both events, should we delete all bookings in the same timeslot & event type that were awaiting payment?